test/check-machines-filesystems

```diff
@@ -0,0 +1,117 @@
+        self.assertEqual('dir1', self.machine.execute(xmllint_element.format(prop='devices/filesystem/target/@dir')).strip())
+        self.assertEqual('on', self.machine.execute(xmllint_element.format(prop='devices/filesystem/binary/@xattr')).strip())
+        self.assertEqual('shared', self.machine.execute(xmllint_element.format(prop='memoryBacking/access/@mode')).strip())
+
+        # Add a new shared fileystem - now the memoryBacking is configured and hidden from the dialog
+        b.click("#vm-subVmTest1-filesystems-add")
+        b.set_file_autocomplete_val("#vm-subVmTest1-filesystems-modal-source-group", "/tmp/dir2/")
+        b.set_input_text("#vm-subVmTest1-filesystems-modal-mountTag", "dir2")
+        b.click(".pf-c-expandable-section__toggle")
+        b.set_checked("#vm-subVmTest1-filesystems-modal-xattr", False)
+        b.wait_not_present("#vm-subVmTest1-filesystems-modal-file-backed")
+
+        b.click("#vm-subVmTest1-filesystems-modal-add")
+        b.wait_visible("tr[data-row-id='vm-subVmTest1-filesystem-/tmp/dir2/-dir2']")
+        b.wait_in_text("tr[data-row-id='vm-subVmTest1-filesystem-/tmp/dir2/-dir2'] td[data-label='Source path']", "/tmp/dir2/")
+        b.wait_in_text("tr[data-row-id='vm-subVmTest1-filesystem-/tmp/dir2/-dir2'] td[data-label='Mount tag']", "dir2")
+
+        self.assertEqual('/tmp/dir2/', self.machine.execute(xmllint_element.format(prop='devices/filesystem[2]/source/@dir')).strip())
+        self.assertEqual('dir2', self.machine.execute(xmllint_element.format(prop='devices/filesystem[2]/target/@dir')).strip())
+        self.assertEqual('', self.machine.execute(xmllint_element.format(prop='devices/filesystem[2]/binary/@xattr')).strip())
+
+        # Try to add a new shared filesystem with the same mount tag
+        # Due to a libvirt bug this will be actually added without an error
+        # Keep the test here and add the expected behavior when the BZ is fixed
+        # https://bugzilla.redhat.com/show_bug.cgi?id=1972218
+        b.click("#vm-subVmTest1-filesystems-add")
+        b.set_file_autocomplete_val("#vm-subVmTest1-filesystems-modal-source-group", "/tmp/dir1/")
+        b.set_input_text("#vm-subVmTest1-filesystems-modal-mountTag", "dir1")
+        b.click("#vm-subVmTest1-filesystems-modal-add")
+        b.wait_visible("tr:nth-child(3)[data-row-id='vm-subVmTest1-filesystem-/tmp/dir1/-dir1']")
+
+        # Try to add a new shared filesystem with non existing source directory
+        b.click("#vm-subVmTest1-filesystems-add")
+        b.set_input_text("#vm-subVmTest1-filesystems-modal-source-group input", "dir3")
+        b.wait_in_text("#vm-subVmTest1-filesystems-modal-source", "No such file or directory")
+        b.click("#vm-subVmTest1-filesystems-modal-cancel")
+
+        # Start VM and ensure that adding filesystem is disabled
+        b.click("#vm-subVmTest1-run")
+        b.wait_visible("#vm-subVmTest1-filesystems-add[aria-disabled=true]")
+
```

Comment:
Would be good to check that the card/button don't appear on a session VM. This works right now, but let's make sure it stays that way. :-)

Reply:
Will send shortly a follow up. Just want to make sure this is in before the end of the day :) And tests more people can review!